hal/halx86/CMakeLists.txt

```diff
@@ -38,6 +38,15 @@ function(add_hal _halname)
     else()
         target_link_libraries(${_halname} -lgcc)
     endif()
+
+    if(SANITIZE_UB OR KASAN_ENABLED)
+        target_link_libraries(${_halname} ksanitize)
+        if(KASAN_ENABLED)
+            target_compile_options(${_halname} PRIVATE "-fsanitize=address")
+        else()
+            target_compile_options(${_halname} PRIVATE "-fsanitize=undefined;-fno-sanitize=alignment")
```

Comment:
Why is this needed for the hals? This seemed to not be necessary for bootvid or kdrosdbg :thinking:

Reply:
I saw that Viktor linked ksanitize here for ubsan, so I thought we should for asan as well. I can remove if needed